### PR TITLE
Added Light class & PerFrameDrawing system

### PR DESCRIPTION
**Pull request by kagikn**

### CODE REVIEW

**Comment by crosire** [comment]
May I ask, what is it the PerFrameDrawing event can do better than the Tick event?

**Comment by kagikn** [comment]
@crosire Yep, PerFrameDrawing can do regardness of the Interval.

**Comment by crosire** [comment]
Alright. But do we really need it? If users need that they can always instantiate a second script class with an interval of zero for drawing stuff.

File: source/Light.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#include "Script.hpp"
+#include "Native.hpp"
+
+namespace GTA
+{
+	using namespace System;
+
+	Light::Light(Math::Vector3 position, Drawing::Color color, float range, float intensity)
+	{
+		_isEnabled = false;
```
Comment (JohnnyCrazy) [inline source/Light.cpp]: Shouldn't the light be initially enabled? When creating a new object, lot of users would expect it being initialized. (just my thoughts)  Maybe another constructor overload would be good

**Comment by kagikn** [comment]
As you may know, Light class is implemented in GTAIV ScriptHookDotNet.

**Comment by timnboys** [comment]
Please I ask @crosire @JohnnyCrazy please add this class as this is the finishing part of my script I need to get my script ported over from GTA-IV ScripthookDotNet to GTAV's ScriptHookDotNet.
to allow me to draw textures like they worked in gta iv.

**Pull request closed without merging**